### PR TITLE
Adapted IInput And Socket To New Design.

### PR DESCRIPTION
**Pull request by TomerGibor**

### CODE REVIEW

File: Dev/VideoChatClient/Common/Common.hpp [hunk 1]
```diff
@@ -4,3 +4,22 @@
 #include <vector>
 
 using Buffer = std::vector<uint8_t>;
+
+
+enum class PacketDataTypes : uint16_t
+{
+    UninitDataType = 0,
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Common/Common.hpp:11]: Uninitialized

File: Dev/VideoChatClient/Common/Common.hpp [hunk 2]
```diff
@@ -4,3 +4,22 @@
...
+	TextDataType,
+	ImageDataType,
+	AudioDataType,
+};
+
+struct PacketHeaders            // 80 bytes
+{
+    char RT_Magic[2];           // 2 bytes
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Common/Common.hpp:19]: Magic numbers? Not in our code!🌟

File: Dev/VideoChatClient/Common/Common.hpp [hunk 3]
```diff
@@ -4,3 +4,22 @@
...
+    uint32_t meeting_id;        // 4 bytes
+    uint32_t user_unique_id;    // 4 bytes
+    char name[64];              // 64 bytes
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Common/Common.hpp:22]: Use std::array of this size, magic number here too

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 4]
```diff
@@ -8,7 +8,8 @@
 class IInput
 {
 public:
-	IInput() = default;
+	IInput() : m_input_type(PacketDataTypes::UninitDataType) {}
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:11]: Remove this ctor, we do not support uninitialized type

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 5]
```diff
@@ -8,7 +8,8 @@
...
+	IInput(PacketDataTypes input_type) : m_input_type(input_type) {}
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:12]: Curly braces at the line below

File: Dev/VideoChatClient/Input/ImageInput.cpp [hunk 6]
```diff
@@ -2,7 +2,9 @@
 
 #include "VideoChatClientException/VideoChatClientException.hpp"
 
-ImageInput::ImageInput() : m_capture()
+ImageInput::ImageInput() :
+	IInput(PacketDataTypes::ImageDataType),
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/ImageInput.cpp:6]: Good

File: Dev/VideoChatClient/Input/TextInput.cpp [hunk 7]
```diff
@@ -1,5 +1,7 @@
 #include "TextInput.hpp"
 
+TextInput::TextInput() : IInput(PacketDataTypes::TextDataType)
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/TextInput.cpp:3]: Each line at the Initialization list should be in separate line

Comment (ron-elmekiesse) [reply Dev/VideoChatClient/Input/TextInput.cpp:3]: Fix everywhere

**Review by ron-elmekiesse — COMMENTED** [review]
I did the review from my phone, hoped I recovered everything well

File: Dev/VideoChatClient/Common/Common.hpp (hunk 3 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Common/Common.hpp:22]: We can't do that, since we are writing to it directly so we can't use a class

**Pull request merged**